core/shared/src/main/scala/org/http4s/headers/Proxy-Authorization.scala

```diff
@@ -33,8 +34,8 @@ object `Proxy-Authorization` {
     import org.http4s.internal.parsing.AuthRules.credentials
     credentials.map(`Proxy-Authorization`(_))
   }
-  def apply(basic: BasicCredentials): Authorization =
```

Comment:
`Authorization` -> `Proxy-Authorization`
Isn't a binary-compatible change, and usually we make such changes only in the `1.0` series, but not in `0.23`.
So I'd suggest marking this `apply` as `deprecated` and introducing the `fromBasicCredentials` or simply `from`.